### PR TITLE
docs: Add Valhalla docs

### PR DESCRIPTION
**Pull request by shirCitron**

### CODE REVIEW

File: docs/MapColonies/vector/Services/valhalla_service.md
Comment (CptSchnitz) [inline docs/MapColonies/vector/Services/valhalla_service.md:1]: 1. The order of the page could be better. it should be about selling the service so terminology should not be first (maybe even on a separate page).
2. The page is missing some pictures to sell the service.

File: docs/MapColonies/vector/Services/valhalla_service.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+5. Includes turn restrictions, highway exit information, administrative information, and timezones.
+
+## How to use
+
+Valhalla is an open-source toolkit for multimodal transportation, powered by open data. The APIs use the standard REST model of interaction with JSON serving as both the request and response formats. Please refer to a specific service's API documentation to learn about the various request/response properties. In addition to JSON request/response formats Valhalla also supports protocol buffers as request/response format.
+
+Its API has the following endpoints for querying the data:
+* __[route](/docs/MapColonies/openstreetmap/valhalla/turn-by-turn/osm-valhalla-route)__ - the service guides you between points by car, bike, foot, and multimodal combinations involving walking and riding public transit. Your apps can use the results from the route service to plan multimodal journeys with narratives to guide users by text and by voice. Valhalla draws data from OpenStreetMap for the main graph and from user-supplied GTFS feeds for multimodal routing. 
+    
+    #### "I am trying to run more than one errand in the day or start my own delivery service!" 🤷
```
Comment (CptSchnitz) [inline docs/MapColonies/vector/Services/valhalla_service.md:51]: why is there a title in the middle of a bulleted list?
image

File: docs/MapColonies/openstreetmap/valhalla/turn-by-turn/api-reference.md [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+|441 | Location is unreachable |
+|442 | No path could be found for input |
+|443 | Exact route match algorithm failed to find path |
+|444 | Map Match algorithm failed to find path |
+|445 | Shape match algorithm specification in api request is incorrect. Please see documentation for valid shape_match input. |
+|499 | Unknown |
+|**5xx** | **Tyr project codes** |
+|500 | Failed to parse intermediate request format |
+|501 | Failed to parse TripDirections |
+|599 | Unknown |
```
Comment (CptSchnitz) [inline docs/MapColonies/openstreetmap/valhalla/turn-by-turn/api-reference.md:623]: should not be part of the developer portal

File: docs/MapColonies/openstreetmap/valhalla/turn-by-turn/api-reference.md
Comment (CptSchnitz) [inline docs/MapColonies/openstreetmap/valhalla/turn-by-turn/api-reference.md:1]: something is bad in the title hierarchy. i don't think `Directions options` should be under `Costing models`.

**Review by CptSchnitz — CHANGES_REQUESTED** [review]
I think there are too many APIs and a lack of examples. we need to somehow make the docs more concise.

File: docs/MapColonies/vector/Services/valhalla_service.md [hunk 3]
```diff
@@ -43,18 +50,16 @@ Valhalla is an open source routing engine and accompanying libraries for use wit
...
-* __[isochrone](/docs/MapColonies/openstreetmap/valhalla/osm-valhalla-isochrone)__ - get a computation of areas that are reachable within specified time periods from a location or set of locations
-* __[map-matching](/docs/MapColonies/openstreetmap/valhalla/osm-valhalla-map-matching)__ - matches coordinates to known roads so you can turn a path into a route with narrative instructions and get the attribute values from that matched line. 
+* __[isochrone](/docs/MapColonies/openstreetmap/valhalla/osm-valhalla-isochrone)__ - get a computation of areas that are reachable within specified time periods from a location or set of locations.
 * __[expansion](/docs/MapColonies/openstreetmap/valhalla/osm-valhalla-expansion)__ - return a geojson representation of a graph traversal at a given location.
 * __[locate](/docs/MapColonies/openstreetmap/valhalla/osm-valhalla-locate)__ - get detailed metadata about the nodes and edges in the graph.
 
 * **centroid** - allows you to find the least cost convergence point of routes from multiple locations.
```
Comment (CptSchnitz) [inline docs/MapColonies/vector/Services/valhalla_service.md:62]: if we have no documentation about it, just remove the line.

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

**Comment by shimoncohen** [comment]
Please move all of `OPenStreetMap` under `Knowledge base` under `Vector`.
In addition, I think we should move all of the API's to be under the service description, what do you think?

**Comment by shirCitron** [comment]
>

File: docs/MapColonies/openstreetmap/valhalla/turn-by-turn/overview.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+---
+
+# Routing overview
+
+The Valhalla route service (a.k.a. turn-by-turn) is an open-source routing service that lets you integrate routing and navigation into a web or mobile application. The service works globally, and provides dynamic and customizable routing by driving, walking, bicycling, and using multimodal and transit options, with clear directions for maneuvers along the route.
+
+![Routing settings](../../../../../static/img/openstreetmap/valhalla/routing-settings.png)
+-
+![Routing settings](../../../../../static/img/openstreetmap/valhalla/routing-transport.png)
```
Comment (shimoncohen) [inline docs/MapColonies/openstreetmap/valhalla/turn-by-turn/overview.md:20]: Did you mean to write `Routing transport`?

File: docs/MapColonies/openstreetmap/valhalla/turn-by-turn/overview.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+The service [route results](api-reference.md#outputs-of-a-route) provide details about the trip, including locations, a summary with basic information about the entire trip and a list of legs. Each leg has its own summary, a shape, which is an encoded polyline of the route path, and a list of maneuvers. These maneuvers provide written narrative instructions, plus verbal alerts that can be used as audio guidance in navigation apps.
+
+The JSON returned from the route query can be drawn on a map and shown as instructions for maneuvers along the route.
+
+## Examples
+
+![Routing settings](../../../../../static/img/openstreetmap/valhalla/route-example.png)
```
Comment (shimoncohen) [inline docs/MapColonies/openstreetmap/valhalla/turn-by-turn/overview.md:32]: Did you mean to write `Route example`?

File: docs/MapColonies/vector/Services/valhalla_service.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+description: Directions and isochrones by OSM
+tags:
+  - vector
+  - API
+  - osm
+---
+
+# Valhalla 🧭
+
+Valhalla is an open source routing engine and accompanying libraries for use with OpenStreetMap data. Valhalla also includes tools like time+distance matrix computation, isochrones, elevation sampling, map matching and tour optimization (Travelling Salesman).
```
Comment (shimoncohen) [inline docs/MapColonies/vector/Services/valhalla_service.md:14]: We should add a link to the `Valhalla` docs and website to credit them.
Same for `Nominatim`.

**Review by shimoncohen — CHANGES_REQUESTED** [review]

**Review by shimoncohen — APPROVED** [review]

**Review by shimoncohen — DISMISSED** [review]

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**